src/event.rs

```diff
@@ -555,10 +560,33 @@ pub enum KeyEventKind {
 
 bitflags! {
     /// Represents extra state about the key event.
+    ///
+    /// **Note:** This state can only be read if
+    /// [`KeyboardEnhancementFlags::DISAMBIGUATE_ESCAPE_CODES`] has been enabled with
+    /// [`PushKeyboardEnhancementFlags`].
     #[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
     pub struct KeyEventState: u8 {
         /// The key event origins from the keypad.
         const KEYPAD = 0b0000_0001;
+        /// The Hyper modifier was held for this key event.
+        ///
+        /// This modifier is uncommon; thus, it is not stored in `KeyEvent.state` to avoid
+        /// confusion or failures to match key events.
+        const MODIFIER_HYPER = 0b0000_0010;
```

Comment:
I've always heard hyper to be the combination of all mod keys i.e. all of shift, control, alt, and super pressed at once. See https://docs.folivora.ai/docs/1004_hyper_key.html or https://kaleidoscope.readthedocs.io/en/latest/NEWS.html#hyper-and-meh-keys for examples of that interpretation. https://askubuntu.com/questions/19558/what-are-the-meta-super-and-hyper-keys shows where it used to exist on an actual keyboard, but these modern usages have muddied the water.

Given the lack of modern keyboard support for hyper and meta, I'd vote for not including these. Having some mods be in KeyEventState and some in KeyModifiers feels confusing to me.

Reply:
I'd be fine with that. Didn't love my solution either. @TimonPost ?

Reply:
If they are uncommon you could leave them out. I never used them but that should not be the leading factor. I agree it might be confusing to have modifier keys at two places. Also" "to avoid confusion or failures to match key events.", is this a real problem we face? In the sense will people not reading those keys have a harder time if it is within the `state` field?

Reply:
Maybe just add them to the `KeyModifiers` even if they are uncommen.

Reply:
Will do. I'd prefer that to having them in the state.